src/library/scala/runtime/ModuleSerializationProxy.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.runtime
+
+import java.io.Serializable
+import java.security.PrivilegedActionException
+import java.security.PrivilegedExceptionAction
+
+private[runtime] object ModuleSerializationProxy {
+  private val instances = new ClassValueCompat[Object] {
+    override protected def computeValue(`type`: Class[_]): Object = {
+      try {
+        java.security.AccessController.doPrivileged(new PrivilegedExceptionAction[Object] {
+          override def run(): Object = {
+            `type`.getField("MODULE$").get(null)
+          }
+        })
+      } catch {
+        case e: PrivilegedActionException =>
+          rethrowRuntime(e.getCause)
+      }
+    }
+  }
+
+  private def rethrowRuntime(e: Throwable): Object = {
+    val cause = e.getCause
```

Comment:
Twice `.getCause` seems a bit suspicious (one in the definition of `rethrowRuntime` and one when it is used). If it's intentional, it would benefit from a comment explaining the reason. Thank you! :smiley: